katello/hooks/pre_commit/13-hiera.rb

```diff
@@ -7,11 +7,23 @@
   'extra-extra-large' => { cpu_cores: 48, memory: 256 },
 }.freeze
 TUNING_FACT = 'tuning'.freeze
+MONGO_CACHE_SIZE_FACT = 'mongo_cache_size'.freeze
 
 EXIT_INVALID_TUNING = 101
 EXIT_INSUFFICIENT_CPU_CORES = 102
 EXIT_INSUFFICIENT_MEMORY = 103
 
+# Setting Default Mongo WiredTiger Engine cache size to be 20% of total memory
+# See https://access.redhat.com/solutions/4505561
+# See https://docs.mongodb.com/manual/core/wiredtiger/#memory-use
+current_mongo_cache_size = get_custom_fact(MONGO_CACHE_SIZE_FACT)
+required_mongo_cache_size = (facts[:memory][:system][:total_bytes] * 0.2 / 1024 / 1024 / 1024).round(2)
+if current_mongo_cache_size != required_mongo_cache_size
+  store_custom_fact(MONGO_CACHE_SIZE_FACT, required_mongo_cache_size)
+  # Store the app config to disk
+  kafo.config.configure_application
```

Comment:
Now that I think about this, this is something we do wrong in the other one as well I think but it should respect noop
```suggestion
  kafo.config.configure_application unless app_value(:noop)
```

Reply:
Ok, it's actually time for weekend. Kafo itself already handles that and there's no need to worry about it here.

Reply:
ok great thank you for the info.